lib/fileExportReporter.js

```diff
@@ -68,23 +80,37 @@ const FileExportReporter = function(baseReporterDecorator, config, logger) {
 			return;
 		}
 
-		for (const file of result.exportFiles) {
-			const pathSegments = [outputDir, path.basename(file.name)];
-			if (multiBrowsers) {
-				pathSegments.splice(1, 0, path.basename(browser.name));
-			}
+		if (!Array.isArray(result.exportFiles)) {
+			log.warn("Export files must be given as an array");
+			return;
+		}
+
+		let exportPath = outputDir;
+		if (multiBrowsers) {
+			exportPath = path.join(exportPath, escapeFileName(browser.name));
+		}
+		try {
+			for (const file of result.exportFiles) {
+				if (typeof file.name !== "string" || typeof file.content !== "string") {
+					log.warn("Invalid file object. \"name\" and \"content\" must be strings");
+					continue;
+				}
 
-			await writeSingleFile(path.join.apply(null, pathSegments), file.content);
+				await writeSingleFile(exportPath, escapeFileName(file.name), file.content);
+			}
+		} catch (err) {
+			log.warn("An unexpected error occured while exporting files\n\t" + err.message);
```

Comment:
if we set exit code 1, we should also log this as an error

Reply:
Done